File: assignments/english_digit_to_myanmar_digit/python/english_digit_to_myanmar_digit_test_runner.py
import sys
from english_digit_to_myanmar_digit import EnglishDigitToMyanmarDigit

def run_tests():
    solver = EnglishDigitToMyanmarDigit()
    test_cases = [
        (0, "၀"), (9, "၉"), (10, "၁၀"), (11, "၁၁"), (1234, "၁၂၃၄"),
        (2314, "၂၃၁၄"), (987654321, "၉၈၇၆၅၄၃၂၁")
    ]
    passed_count = 0
    report = []
    print("===== Starting Automated Tests for EnglishDigitToMyanmarDigit (Python) =====")

    for i, (input_val, expected_output) in enumerate(test_cases):
        try:
            actual_output = solver.convert(input_val)
            report.append(f"Test Case #{i + 1}: Input '{input_val}'")
            report.append(f" -> Expected: \"{expected_output}\", Got: \"{actual_output}\"")

            if actual_output == expected_output:
                report.append(" -> ✅ PASS\n")
                passed_count += 1
            else:
                report.append(" -> ❌ FAIL\n")
        except Exception as e:
            report.append(f" -> ❌ ERROR: An exception occurred - {type(e).__name__}\n")

    print("\n".join(report))
    print("===== Test Summary ====="); print(f"Total Tests: {len(test_cases)}"); print(f"Passed: {passed_count}"); print(f"Failed: {len(test_cases) - passed_count}"); print("========================")
    print(f"[SUMMARY]:{passed_count}:{len(test_cases)}")

if __name__ == "__main__":
    sys.path.append('.')
    run_tests()

File: assignments/english_digit_to_myanmar_digit/javascript/english_digit_to_myanmar_digit_test_runner.js
const assert = require('assert');
const { EnglishDigitToMyanmarDigit } = require('./EnglishDigitToMyanmarDigit.js');

function runTests() {
    const solver = new EnglishDigitToMyanmarDigit();
    const testCases = [
        { input: 0, expected: "၀" }, { input: 9, expected: "၉" },
        { input: 10, expected: "၁၀" }, { input: 11, expected: "၁၁" },
        { input: 1234, expected: "၁၂၃၄" }, { input: 2314, expected: "၂၃၁၄" },
        { input: 987654321, expected: "၉၈၇၆၅၄၃၂၁" }
    ];

    let passedCount = 0;
    let report = [];
    console.log("===== Starting Automated Tests for EnglishDigitToMyanmarDigit (JavaScript) =====");

    testCases.forEach((tc, i) => {
        let actual = "";
        try {
            actual = solver.convert(tc.input);
            assert.strictEqual(actual, tc.expected);
            report.push(`Test Case #${i + 1}: Input ${tc.input} -> ✅ PASS`);
            passedCount++;
        } catch (error) {
             report.push(`Test Case #${i + 1}: Input ${tc.input} -> ❌ FAIL/ERROR: Expected "${tc.expected}", but got "${actual}"`);
        }
    });

    console.log(report.join('\n\n'));
    console.log("\n===== Test Summary =====");
    console.log(`Total Tests: ${testCases.length}`); console.log(`Passed: ${passedCount}`); console.log(`Failed: ${testCases.length - passedCount}`); console.log("========================");
    console.log(`[SUMMARY]:${passedCount}:${testCases.length}`);
}

runTests();